iina/AppDelegate.swift

```diff
@@ -584,10 +584,56 @@ class AppDelegate: NSObject, NSApplicationDelegate, SPUUpdaterDelegate {
     let urls = pendingFilesForOpenFile.map { URL(fileURLWithPath: $0) }
 
     pendingFilesForOpenFile.removeAll()
-    if PlayerCore.activeOrNew.openURLs(urls) == 0 {
+    
+    if urls.count == 1,
+       let url = urls.first,
+       url.pathExtension == "iinaplgz" {
+      installPlugin(form: url)
+    } else if PlayerCore.activeOrNew.openURLs(urls) == 0 {
       Utility.showAlert("nothing_to_open")
     }
   }
+  
+  // MARK: - Plugin Installer
+  func installPlugin(form url: URL) {
+    do {
+      let plugin = try JavascriptPlugin.create(fromPackageURL: url)
+      
+      let previousPlugin = JavascriptPlugin.plugins.first {
+        $0.identifier == plugin.identifier
+      }
+      
+      func updatePlugin() {
+        Utility.showPermissionsAlert(forPlugin: plugin,
+                                     previousPlugin: previousPlugin) { ok in
+          guard ok else {
+            plugin.remove()
+            return
+          }
+          
+          if let previousPlugin,
```

Comment:
This statement makes use of [SE-0345](https://github.com/apple/swift-evolution/blob/main/proposals/0345-if-let-shorthand.md), the "if let shorthand", implemented in Swift 5.7 which is supported by Xcode 14. IINA's [CONTRIBUTING](https://github.com/iina/iina/blob/develop/CONTRIBUTING.md) guildlines recommends using the latest Xcode, so this submission is correct and proper. Except… IINA is documented as supporting OS X 10.11+ and Xcode 14 dropped support for 10.11 and 10.12.

Last I heard the plan was to continue to support 10.11 and 10.12 for this next release by building with Xcode 13. This is why there are two CI jobs, one with Xcode 13 and one with Xcode 14. Since Xcode 13.4 only supports Swift 5.6 the first CI build failed with compilation errors on statements like this one.